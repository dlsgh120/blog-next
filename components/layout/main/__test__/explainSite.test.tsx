import React from 'react';
import {render} from '@testing-library/react';
import ExplainSite from '../explainSite';

describe('<ExplainSite />',()=>{
    it('<ExplainSite /> snapshot', ()=>{
        const el = render(<ExplainSite />);
        expect(el).toMatchSnapshot();
    });

    it('has a text of <h1> from ExplainContainer', ()=>{
        const el = render(<ExplainSite />);
        const h1 = el.getByText('Try the various services provided by Choi Tech.');
        expect(h1).toBeTruthy();
    });

    it('has a text of <h2> from ExplainContent', ()=>{
        const el = render(<ExplainSite />);
        const h2 = el.getByText('지식 공유');
        expect(h2).toBeTruthy();
    })
    
    it('has a text of <p> from ExplainContent', ()=>{
        const el = render(<ExplainSite />);
        const p = el.getByText('Front-end 프레임워크인 React에 관련된 기술 및 경험을 공유 합니다. 주로 개발 환경 구성 및, 이슈 대응, 문제 해결 등 다양한 주제를 다루고 있습니다. 또한, 공유한 지식에 대하여, 질의 응답 등을 이용 할 수 있습니다.');
        expect(p).toBeTruthy();
    })

})