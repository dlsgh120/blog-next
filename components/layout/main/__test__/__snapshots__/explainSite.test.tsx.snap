// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExplainSite /> <ExplainSite /> snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="explainSite__ExplainWrap-twxkik-0 bYGHGQ"
      >
        <div
          class="explainSite__ExplainContainer-twxkik-1 dRaxaZ"
        >
          <h1>
            Try the various services provided by Choi Tech.
          </h1>
          <div
            class="explainSite__ExplainContent-twxkik-2 iDmgc"
          >
            <div>
              <h2>
                지식 공유
              </h2>
              <p>
                Front-end 프레임워크인 React에 관련된 기술 및 경험을 공유 합니다. 주로 개발 환경 구성 및, 이슈 대응, 문제 해결 등 다양한 주제를 다루고 있습니다. 또한, 공유한 지식에 대하여, 질의 응답 등을 이용 할 수 있습니다.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="explainSite__ExplainWrap-twxkik-0 bYGHGQ"
    >
      <div
        class="explainSite__ExplainContainer-twxkik-1 dRaxaZ"
      >
        <h1>
          Try the various services provided by Choi Tech.
        </h1>
        <div
          class="explainSite__ExplainContent-twxkik-2 iDmgc"
        >
          <div>
            <h2>
              지식 공유
            </h2>
            <p>
              Front-end 프레임워크인 React에 관련된 기술 및 경험을 공유 합니다. 주로 개발 환경 구성 및, 이슈 대응, 문제 해결 등 다양한 주제를 다루고 있습니다. 또한, 공유한 지식에 대하여, 질의 응답 등을 이용 할 수 있습니다.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
